### PR TITLE
Bug 683 fix login issues

### PR DESCRIPTION
**Pull request by danielsh28**

### CODE REVIEW

File: src/components/atoms/Button.tsx [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-interface IProps {
+interface IProps extends ButtonProps {
   onClick?: () => any;
 }
 
 // todo: make standard buttons for entire app
-const StandardButton: FC<IProps> = ({ onClick, children }) => (
-  <MatButton variant="contained" color="primary" onClick={onClick}>
+const StandardButton: FC<IProps> = ({ type, onClick, children }) => (
+  <MatButton type={type} variant="contained" color="primary" onClick={onClick}>
```
Comment (yuvalbl) [inline src/components/atoms/Button.tsx:10]: We prefer specific buttons over generic ones.
see my post about it
https://medium.com/@yuval.bar.levi/react-spread-attributes-dont-overuse-them-3f322a0ace22

File: package.json [hunk 2]
```diff
@@ -25,6 +25,7 @@
     "react": "^16.14.0",
     "react-async-loader": "^0.1.2",
     "react-dom": "^16.14.0",
+    "react-hook-form": "^7.8.1",
```
Comment (yuvalbl) [inline package.json:28]: why adding react-hook-form if you're not using it?

File: src/components/atoms/Button.tsx [hunk 3]
```diff
@@ -1,13 +1,13 @@
 import React, { FC } from 'react';
-import { Button as MatButton, IconButton as MatIconButton } from '@material-ui/core';
+import { Button as MatButton, ButtonProps, IconButton as MatIconButton } from '@material-ui/core';
 
-interface IProps {
+interface IProps extends ButtonProps {
```
Comment (yuvalbl) [inline src/components/atoms/Button.tsx:4]: I rather not allowing all these props. see my comment below

File: .env.production [hunk 4]
```diff
@@ -1,6 +1,6 @@
 # override .env
-REACT_APP_REDIRECT_URL =https://anyway-infographics.web.app/login-popup-redirect
-
+REACT_APP_REDIRECT_URL =/login-popup-redirect
+REACT_APP_ORIGIN=https://anyway-infographics.web.app
```
Comment (yuvalbl) [inline .env.production:3]: use `location.origin` instead

File: src/utils/utils.ts [hunk 5]
```diff
@@ -29,12 +29,13 @@ export const serverUrl = process.env.REACT_APP_BASE_URL;
...
 export const AUTH_LOGIN_GOOGLE_URL: URL = new URL(`${authServerUrl}authorize/google`);
-export const REDIRECT_URL: string | undefined = process.env.REACT_APP_REDIRECT_URL;
+export const REDIRECT_ROUTE: string | undefined = process.env.REACT_APP_REDIRECT_URL;
+export const ORIGIN: string | undefined = process.env.REACT_APP_ORIGIN;
 export const GET_USER_INFO_URL = `${authServerUrl}user/info`;
 export const UPDATE_USER_INFO_URL = `${authServerUrl}user/update`;
 export const LOG_OUT_USER_URL = `${authServerUrl}logout`;
 
-AUTH_LOGIN_GOOGLE_URL.searchParams.append('redirect_url', REDIRECT_URL!);
+AUTH_LOGIN_GOOGLE_URL.searchParams.append('redirect_url', `${ORIGIN}${REDIRECT_ROUTE!}`);
```
Comment (yuvalbl) [inline src/utils/utils.ts:38]: `location.origin`

File: .env.production [hunk 6]
```diff
@@ -1,6 +1,6 @@
 # override .env
-REACT_APP_REDIRECT_URL =https://anyway-infographics.web.app/login-popup-redirect
-
+REACT_APP_REDIRECT_URL =/login-popup-redirect
```
Comment (yuvalbl) [inline .env.production:2]: `REACT_APP_REDIRECT_URL` is the same - can be removed?

File: .env.staging [hunk 7]
```diff
@@ -1,5 +1,6 @@
 # override .env, .env.production
-REACT_APP_REDIRECT_URL=https://anyway-infographics-staging.web.app/login-popup-redirect
+REACT_APP_REDIRECT_URL=/login-popup-redirect
+REACT_APP_ORIGIN=https://anyway-infographics-staging.web.app
```
Comment (yuvalbl) [inline .env.staging:3]: can be removed - same as above

File: src/components/atoms/Button.tsx [hunk 8]
```diff
@@ -3,11 +3,12 @@ import { Button as MatButton, IconButton as MatIconButton } from '@material-ui/c
 
 interface IProps {
   onClick?: () => any;
+  type?: 'button' | 'submit' | 'reset';
```
Comment (yuvalbl) [inline src/components/atoms/Button.tsx:6]: replace with: `isSubmit`

File: src/components/atoms/Button.tsx (hunk 1 above)
Comment (yuvalbl) [reply src/components/atoms/Button.tsx:11]: replace with :
```
const buttonType = {}
if(isSubmit) {
buttonType.type = 'submit';
}
<MatButton {...buttonType} variant="contained" color="primary" onClick={onClick}>
```

File: src/components/molecules/UserUpdateForm.tsx [hunk 9]
```diff
@@ -31,12 +32,25 @@ const useStyles = makeStyles((theme) => ({
     justifyContent: 'center',
     marginBottom: theme.spacing(2),
   },
+  show: {
+    visibility: 'visible',
+  },
+  hide: {
+    visibility: 'hidden',
+  },
```
Comment (yuvalbl) [inline src/components/molecules/UserUpdateForm.tsx:40]: no need for both

File: src/utils/utils.ts [hunk 10]
```diff
@@ -30,7 +30,7 @@ export const serverUrl = process.env.REACT_APP_BASE_URL;
 export const authServerUrl = process.env.REACT_APP_AUTH_URL;
 export const AUTH_LOGIN_GOOGLE_URL: URL = new URL(`${authServerUrl}authorize/google`);
 export const REDIRECT_ROUTE: string | undefined = process.env.REACT_APP_REDIRECT_URL;
-export const ORIGIN: string | undefined = process.env.REACT_APP_ORIGIN;
+export const ORIGIN: string | undefined = window.location.origin;
```
Comment (yuvalbl) [inline src/utils/utils.ts:33]: not required. simply use ``window.location.origin` where needed

**Review by yuvalbl — APPROVED** [review]

**Review by almogbhl — APPROVED** [review]

**Pull request merged**